### ISSUE TITLE
How is SingleSelectFilter supposed to work?

### ISSUE THREAD

**Issue by NicoleRauch** [adazzle/react-data-grid#1463]
### Which version of React JS are you using?

✅ Officially supported ✅
- [ ] v15.4.x

⚠️ Not officially supported, expect warnings ⚠️
- [ ] v15.5.x
- [ ] v15.6.x

☣️ Not officially supported, expect warnings and errors ☣️
- [X] v16.x.x

---

### Which browser are you using?

✅ Officially supported ✅
- [ ] IE 9 / IE 10 / IE 11
- [ ] Edge
- [X] Chrome

⚠️ Not officially supported, but "should work" ⚠️
- [ ] Firefox
- [ ] Safari

---

### I'm submitting a ...

- [X] 🐛 Bug Report ?
- [ ] 💡 Feature Request

> 👋 Need general support? Not sure about how to use React itself, or how to get started with the Grid?
> Please do not submit support request here. Instead see

> https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md

---

### Issue Details

I am trying to figure out how the SingleSelectFilter works; maybe somebody can shed some light on this?

**My current situation:** I have set up a SingleSelectFilter for a certain column. The column contains rows with the following values:

![image](https://user-images.githubusercontent.com/436499/52533782-4aa19500-2d39-11e9-94df-ebc462757b5f.png)

Now I do not want the filter to contain the combined information "Kategorie 1 | Kategorie 2"; I want it to filter for the individual entries. Therefore I only return the individual entries when asked for the valid filter values:

![image](https://user-images.githubusercontent.com/436499/52533824-d87d8000-2d39-11e9-81f9-b12b82f896c1.png)

-----

Here is what I tried so far and what I would like to understand:

**Experiment 1:** Filter to "Kategorie 1":

![image](https://user-images.githubusercontent.com/436499/52533805-85a3c880-2d39-11e9-801e-26b00f8877e1.png)

Only the first row gets selected => OK, so probably the filter only looks for exact matches?

**Experiment 2:** Filter to "Kategorie 2":

![image](https://user-images.githubusercontent.com/436499/52533829-ec28e680-2d39-11e9-9816-56e856dccedc.png)

Both rows containing "Kategorie 2" get selected => Hm, so probably the filter only matches at the end of a string?

**Experiment 3:** Filter to "orie 1":

![image](https://user-images.githubusercontent.com/436499/52533841-1b3f5800-2d3a-11e9-9876-be593a9dc566.png)

All three rows get selected => Well... all of a sudden the filter matches in the middle and at the end...

Is this behaviour intentional? If so, can somebody explain what the underlying logic might be? Or is ist a bug?

**Bottom line:** Basically, what I want is that filtering for "Kategorie 1" selects the two lines containing that string, and filtering for "Kategorie 2" selects the two other lines containing that string. So I want a "contains" operation that matches my string no matter where the search string is located.
How can I establish this behaviour using the SingleSelectFilter?
